### PR TITLE
using the cv pan gesture to dismiss.

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/UI/Transitions/DismissTransitionController.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@ class DismissTransitionController: NSObject, UIViewControllerAnimatedTransitioni
 
     let interactionController: PanDismissInteractionController?
 
-    var animator: UIViewPropertyAnimator!
+    var animator: UIViewPropertyAnimator?
```
Comment (Benjamindodgson) [inline Shared/UI/Transitions/DismissTransitionController.swift:16]: I figured this couldn't hurt.

File: Shared/UI/Transitions/DismissTransitionController.swift [hunk 2]
```diff
@@ -58,33 +62,18 @@ class DismissTransitionController: NSObject, UIViewControllerAnimatedTransitioni
 
         let finalFrame = toView.convert(toView.bounds, to: containerView)
 
-        let snapshot = MessageContentView()
```
Comment (Benjamindodgson) [inline Shared/UI/Transitions/DismissTransitionController.swift:61]: Just move the view. No need for a snapshot.

File: Shared/UI/Transitions/PanDismissInteractionController.swift [hunk 3]
```diff
@@ -27,26 +27,27 @@ class PanDismissInteractionController: UIPercentDrivenInteractiveTransition {
...
 
-    func initialize(interactionView: UIView) {
-        let panGesture = UIPanGestureRecognizer { [unowned self] (pan) in
-            self.handle(pan: pan)
-        }
-        panGesture.cancelsTouchesInView = true
-        panGesture.delegate = self
-        interactionView.addGestureRecognizer(panGesture)
+    func initialize(collectionView: UICollectionView) {
+        collectionView.panGestureRecognizer.addTarget(self, action: #selector(self.handle(pan:)))
```
Comment (Benjamindodgson) [inline Shared/UI/Transitions/PanDismissInteractionController.swift:31]: use the collection views pan

**Review by MartinJibber — APPROVED** [review]
Cool

**Pull request merged**